### PR TITLE
MRG FIX: fNIRS picks for different types

### PR DESCRIPTION
**Pull request by rob-luke**
#### Reference issue
I was going to implement some new features raised in an issue from #9141 but found this bug along the way. 

#### What does this implement/fix?
When trying to pick a specific fNIRS type the function was returning all fNIRS types. Now it should just return the requested type.

#### Additional information
Once this bug is fixed I will add new functionality in a seperate PR

### CODE REVIEW

File: mne/preprocessing/nirs/tests/test_nirs.py [hunk 1]
```diff
@@ -76,7 +143,48 @@ def test_fnirs_spread_bads(fname):
...
-    print(raw.info['bads'])
     raw = _fnirs_spread_bads(raw)
-    print(raw.info['bads'])
     assert raw.info['bads'] == [raw.ch_names[x] for x in [0, 1, 8, 9]]
+
+
+@testing.requires_testing_data
+@pytest.mark.parametrize('fname', ([fname_nirx_15_2_short, fname_nirx_15_2,
+                                    fname_nirx_15_0]))
+def test_fnirs_channel_naming_and_order_readers(fname):
```
Comment (rob-luke) [inline mne/preprocessing/nirs/tests/test_nirs.py:153]: Here I am adding testing of an existing function before new functionality that was proposed in #9141 is added in a yet to be opened PR.

**Comment by rob-luke** [comment]
I think this is good to go. @agramfort @larsoner @drammock could you please review this

**Review by agramfort — APPROVED** [review]
LGTM

@larsoner I let you merge?

**Comment by larsoner** [comment]
Thanks @rob-luke !

**Pull request merged**